### PR TITLE
Refactor the way GPIO chips in sysfs get discovered

### PR DESCRIPTION
**Pull request by penguinland**
This is in anticipation of adding something similar for PWM chips to fix RSDK-2332. I recommend reviewing this commit-by-commit, though perhaps it's easy enough all together, too.

No changes to functionality are intended, this is just reorganizing things so they're easier to think about and easier to extend in the future.

Tried on an Orin: GPIO functionality appears unchanged.

### CODE REVIEW

**Comment by penguinland** [comment]
I forgot that Susmita was out sick today. Feel better soon!

File: components/board/genericlinux/data.go [hunk 1]
```diff
@@ -96,21 +98,27 @@ func getCompatiblePinDefs(modelName string, boardInfoMappings map[string]BoardIn
 	return pinDefs, nil
 }
 
-func getBoardMapping(pinDefs []PinDefinition,) (map[int]GPIOBoardMapping, error) {
+func getBoardMapping(pinDefs []PinDefinition) (map[int]GPIOBoardMapping, error) {
+	// What we really want is a mapping from the names of GPIO chips to tuples of their sysfs
+	// directory name, base number, and ngpio (line count). However, because Go doesn't have native
+	// support for tuples, we make a separate mapping for each one.
```
[the remark this answers is not in the document]
Comment (kim-mishra) [reply components/board/genericlinux/data.go]: nvm, I see that you did this

**Review by kim-mishra — APPROVED** [review]
lgtm

File: components/board/genericlinux/data.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (penguinland) [reply components/board/genericlinux/data.go]: Heh, that's such a good idea I already did it in a later commit! Very good instincts. :smile:

**Pull request merged**